File: python/179. Largest Number.py
class Solution:
    # @param {integer[]} nums
    # @return {string}
    def largestNumber(self, nums):
        res = "".join(sorted(map(str, nums), cmp = lambda x, y: cmp(y + x, x + y)))
        return res if res[0] != "0" else "0"

File: cpp/151-200/179. Largest Number.cc
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string> strs;
        for(auto num : nums){
        	strs.push_back(to_string(num));
        }
        string res = "";
        sort(strs.begin(), strs.end(), [](string &s1, string &s2){return s1 + s2 > s2 + s1;});
        for(auto s: strs){
        	res += s;
        }
        while(res[0] == '0' && res.size() > 1)
        	res.erase(0, 1);
        return res;
    }
};